### DISCUSSION TITLE
[February 2024] zkSync developer update

### DISCUSSION THREAD

**Discussion by uF4No** [zkSync-Community-Hub/zksync-developers#399] [Announcements]
Greetings devs! Welcome to the zkSync developer update where we showcase different updates about the protocol, developer tools, integrations, tutorials, developer-oriented content and more.

Here are the updates for the last month:

### 👷 👷‍♀️Protocol upgrade

- Optimizations on `keccak` operations.
- `ecrecover` compatibility fix. Now it behaves as on Ethereum.
- We have revamped the fee model with changes on how the costs to close a batch are distributed between all transactions.
- This upgrade sets the foundation for hyperchains configured as validiums, or using externa DA layers.
- We've also increased the number of transactions that can fit into a batch. The results? zkSync is consistently the cheapest L2

![Screenshot 2024-03-05 at 12 55 28](https://github.com/zkSync-Community-Hub/zksync-developers/assets/1957117/4b02fafc-c0c0-410d-92bf-e1ab97530323)

[Source](https://twitter.com/growthepie_eth/status/1762845715452047748)

### 🛠️ Developer tools

- [web3.js zkSync plugin](https://github.com/web3/web3-plugin-zksync): in collaboration with ChainSafe, we've released the first version of `web3-plugin-zksync`, a plugin that extends web3.js with zkSync-specific RPC methods.
- New commands added to zksync-cli to see transaction info from your terminal.
- Foundry alpha: we released the first alpha version of [foundry-zksync](https://github.com/matter-labs/foundry-zksync). Stay tunned for further updates.
- [Brian AI](https://www.brianknows.org/): a new tool to interact with zkSync ChatGPT-style. It allows to send transactions, deploy contracts or query blockchain data.
- Hardhat plugins: multiple fixes and improvements across all plugins.

### 📖 Documentation

- [New guide to install and use `web3-plugin-zksync`.](https://docs.zksync.io/build/tutorials/tooling-guides/web3js.html)
- [Improved the zkSync CLI section of the docs.](https://docs.zksync.io/build/tooling/zksync-cli/getting-started.html)
- All tutorials were updated to use `zksync-ethers` v6

### 🐳 ETHDenver 2024

Here are the highlights from the team during one of the biggest events of the year: 

- [Why ZK is the endgame?](https://youtu.be/ONEkNCI2iyY?feature=shared&t=4114) keynote by Alex Gluchowski
- [Modular Blockchains and Limitless Scaling of Ethereum](https://www.youtube.com/watch?v=GjFLz_CKCCg) by Anthony Rose
- [L1s vs Rollups zkVM vs zkEVM](https://www.youtube.com/watch?v=T3G98kJQMwg) panel with Zac Williamson, Alex Pruden, Brendan Farmer, Alex Gluchowski
- ZK Mini summit was a special event organised by Matter Labs featuring members from different teams building on ZK. Here are the talks:
    - [Reading of ZK Proof Children's Book](https://youtu.be/O7VERaQCxQ4?si=JO5HJa_onK-mDyqy&t=35) by Marcin Michalski
    - [Dumbass, crazy ideas for on-chain dApps](https://youtu.be/O7VERaQCxQ4?si=DHcGmtv-qgURf4yp&t=976) by Nick Villanueva
    - [Fireside chat : ZK Snarks Rule Everything Around Me](https://youtu.be/O7VERaQCxQ4?si=Naj4F9Dl0WDFbBIw&t=2255) panel moderated by Porter Adams
    - [The Good, The Bad, and The Ugly Communities of L2 networks](https://youtu.be/O7VERaQCxQ4?si=yHUasynYZqo_lAWa&t=4898) panel moderated by bxpana
    - [4844 and zkSync](https://youtu.be/O7VERaQCxQ4?si=Lg5ZeiB7JPoPbrhj&t=7319) by Zach Kolodny
- **ZK Quest**: Hundreds of devs building during Buidlweek had access to ZK Quest; a learn and ear platform where users could earn up to $230 by completing quests and code challenges, and earn exclusive swag. We distributed more than 24ETH (~$90K) to more than 1300 developers as rewards for more than 9500 quests completed.
- **ZK Proofs ELI5**: we released [Zero Knowledge Proofs: explained like I’m 5](https://eli5.zksync.io/)], an interactive online book and game that makes ZK available for everyone.

#### 🔗 Stay connected

- Follow [@zkSyncDevs](https://twitter.com/zkSyncDevs) on X
- Join the #dev-general channel in our Discord

Thanks for being part of the zkSync developer community. Your support and contributions are what make this project thrive.

Keep buidling! 👩‍💻🧑🏻‍💻

**Comment by 0tayfun7** [comment]
Good luck and may you increase your work many times more.:)

**Comment by josephrodriguez9** [comment]
thanks

**Comment by dckesso** [comment]
May God give you more knowledge to explore more
